src/main/java/com/example/demo/service/PatientService.java

```diff
@@ -1,60 +1,94 @@
 package com.example.demo.service;
 
+import com.example.demo.exception.PatientAlreadyExistsException;
 import com.example.demo.exception.PatientIllegalArgumentException;
 import com.example.demo.exception.PatientNotFoundException;
 import com.example.demo.mapper.PatientMapper;
 import com.example.demo.model.FullPatientDataDTO;
+import com.example.demo.model.Patient;
 import com.example.demo.model.PatientDTO;
 import com.example.demo.repository.PatientRepository;
 import lombok.RequiredArgsConstructor;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.time.OffsetDateTime;
 import java.util.List;
+import java.util.Objects;
 import java.util.stream.Collectors;
 
 @Service
+@EnableJpaRepositories(basePackages = "com.example.demo.repository")
 @RequiredArgsConstructor
 public class PatientService {
     private final PatientRepository patientRepository;
     private final PatientMapper patientMapper;
 
     public List<PatientDTO> getAllPatients() {
-        return patientRepository.getAllPatients().stream()
+        return patientRepository.findAll().stream()
                 .map(patientMapper::toDTO)
                 .collect(Collectors.toList());
     }
 
     public PatientDTO getPatient(String email) {
-        return patientMapper.toDTO(patientRepository.getPatient(email)
+        return patientMapper.toDTO(patientRepository.findByEmail(email)
```

Comment:
Patient patient = patientRepository.findByEmail(email)
                .orElseThrow(() -> new PatientNotFoundException("Patient with email: %s does not exist.".formatted(email), OffsetDateTime.now()))
                return mapper.toDto(patient);